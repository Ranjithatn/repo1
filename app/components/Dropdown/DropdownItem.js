import React from 'react'
import Icon from '../Icon/Icon';

const DropdownItem = ({
  className,
  text,
  leftIcon,
  rightIcon,
  onClick,
  disabled=false,
}) => {
   return (
    <div className={"dropdown-item" + (className ? " " + className : "")} style={ disabled ? { pointerEvents: "none", color: '#CCC' } : {zIndex:65} } onClick={onClick}>
      { leftIcon && <Icon icon={leftIcon} />}
      { text }
      { rightIcon && <Icon icon={rightIcon} />}
    </div>
   )
}

export default DropdownItem;
