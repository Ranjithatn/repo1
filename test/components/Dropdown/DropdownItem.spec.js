import React from "react";
import toJson from "enzyme-to-json";
import DropdownItem from "../../../app/components/Dropdown/DropdownItem";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";

configure({ adapter: new Adapter() });


describe('<DropdownItem />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <DropdownItem />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
