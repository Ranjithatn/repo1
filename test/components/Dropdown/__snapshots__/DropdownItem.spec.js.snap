// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> should render without crashing 1`] = `
<DropdownItem>
  <div
    className="dropdown-item"
    style={
      Object {
        "zIndex": 65,
      }
    }
  />
</DropdownItem>
`;
